src/wallet/walletdb.cpp

```diff
@@ -434,7 +436,38 @@ ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue,
             ssKey >> id;
             WalletDescriptor desc;
             ssValue >> desc;
+            if (wss.m_descriptor_caches.count(id) == 0) {
+                wss.m_descriptor_caches[id] = DescriptorCache();
+            }
             pwallet->LoadDescriptorScriptPubKeyMan(id, desc);
+        } else if (strType == DBKeys::WALLETDESCRIPTORCACHE) {
+            bool parent = true;
+            uint256 desc_id;
+            uint32_t key_exp_index;
+            uint32_t der_index;
+            ssKey >> desc_id;
+            ssKey >> key_exp_index;
+
+            // if the der_index exists, it's a derived xpub
+            try
```

Comment:
I wanted to leave this open to different caching structures in the future and not have a ton of cache structure specific records like that.

Reply:
I was thinking the other way around: have different record for different things, so it's easy to add new things and wipe / ignore stuff we don't need anymore.

Reply:
More specifically, I've been thinking about a way to unify the descriptor cache structure so that we don't have to have separate records like that. In particular, I've considered setting a bogus derivation index for the parent xpubs so that a single record can be used, as well as a single map within `DescriptorCache` too.